custom_components/ef_ble/eflib/devices/delta3plus.py

```diff
@@ -0,0 +1,123 @@
+from collections.abc import Sequence
+import logging
+from typing import Any
+
+from bleak.backends.device import BLEDevice
+from bleak.backends.scanner import AdvertisementData
+from google.protobuf.message import Message
+
+from ..commands import TimeCommands
+from ..device_properties import Field, ProtobufField, ProtobufListField, UpdatableProps
+from ..devicebase import DeviceBase
+from ..packet import Packet
+from ..pb import pr705_pb2
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class _StatField[T](ProtobufListField[T]):
+    _stat_enum = pr705_pb2.STATISTICS_OBJECT  # type: ignore[attr-defined]
+
+    @property
+    def list_name(self):
+        return "display_statistics_sum"
+
+    def get_list(self, message: Any) -> Sequence[Message]:
+        return message.list_info
+
+    def should_update(self, message: Any) -> bool:
+        return self._stat_enum.Name(message.statistics_object) == self.pb_field_name
+
+    def get_item_value(self, message: Any) -> T | None:
+        return message.statistics_content
+
+
+def _out_power(x):
+    return -round(x, 2) if x != 0 else 0
+
+
+class Device(DeviceBase, UpdatableProps):
+    """Delta 3 Plus"""
+
+    SN_PREFIX = b"P3"
+    NAME_PREFIX = "EF-D3"
+
+    battery_level = ProtobufField[int]("bms_batt_soc")
+
+    ac_input_power = ProtobufField[float]("pow_get_ac_in")
+    ac_input_energy = _StatField[int]("STATISTICS_OBJECT_AC_IN_ENERGY")
```

Comment:
So yeah, if the device can't calculate those fields - you need to get rid of them or prepare a logic to calculate it within the integration, although that could be a bit complicated...